server/lib/transactions.js

```diff
@@ -242,3 +247,126 @@ export async function sum(where) {
   const result = await models.Transaction.findOne({ attributes, where });
   return result.dataValues.total;
 }
+
+const kindStrings = {
+  ADDED_FUNDS: `Added Funds`,
+  BALANCE_TRANSFER: `Balance Transfer`,
+  CONTRIBUTION: `Contribution`,
+  EXPENSE: `Expense`,
+  HOST_FEE: `Host Fee`,
+  HOST_FEE_SHARE: `Host Fee Share`,
+  HOST_FEE_SHARE_DEBT: `Host Fee Share Debt`,
+  PAYMENT_PROCESSOR_COVER: `Cover of Payment Processor Fee`,
+  PLATFORM_TIP: `Platform Tip`,
+  PLATFORM_TIP_DEBT: `Platform Tip Debt`,
+  PREPAID_PAYMENT_METHOD: `Prepaid Payment Method`,
+};
+
+export async function generateDescription(transaction, { req = null, full = false } = {}) {
+  let baseString = 'Transaction',
+    debtString = '',
+    tierString = '',
+    expenseString = '',
+    fromString = '',
+    toString = '';
+
+  if (transaction.isRefund && transaction.RefundTransactionId) {
+    const refundedTransaction = await (req
+      ? req.loaders.Transaction.byId.load(transaction.RefundTransactionId)
+      : models.Transactions.findByPk(order.RefundTransactionId));
+    if (refundedTransaction) {
+      const refundedTransactionDescription = await generateDescription(refundedTransaction, { req, full });
+      return `Refund of \`${refundedTransactionDescription}\``;
+    }
+  }
+
+  let order, expense, subscription, tier;
```

Comment:
These variables are never reassigned, I would rather use `const` and assign them whenever needed.

Reply:
Uhm, that's not possible without code change regarding variable scopes and re-assigment.